src/preact/utils/url.js

```diff
@@ -0,0 +1,47 @@
+import {
+  assertHttpsUrl,
+  isProtocolValid,
+  // eslint-disable-next-line local/no-forbidden-terms
+  parseUrlWithA,
+  // eslint-disable-next-line import/no-restricted-paths
+} from '../../url';
+
+export const urlUtils = {
```

Comment:
Ditto, this'll need to be scoped per document.

Reply:
https://github.com/ampproject/amphtml/issues/37465